documentation/src/main/asciidoc/topics/proc_configuring_textbased_encoding.adoc

```diff
@@ -0,0 +1,25 @@
+[id='using_text_based_storage-{context}']
+= Using Text-Based Formats for Data
+You can configure {brandname} to encode data in text-based formats such as
+`text/plain`, `application/json`, or `application/xml`.
+
+.Procedure
+
+. Specify a text-based storage format as the MediaType for keys and values.
+. Optionally specify a character set such as `UTF-8`.
++
+The following example configures {brandname} to store entries with the `text/plain; charset=UTF-8` format:
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/encoding_text_plain.xml[]
+----
++
+. Configure your clients.
+
+Hot Rod clients should use the `org.infinispan.commons.marshall.StringMarshaller` marshaller to handle data in text-based formats.
```

Comment:
@gustavonalle is this still correct for Hot Rod clients or should it change to ProtoStream?

Reply:
@ryanemerson same question here, please. for text-based formats should we change StringMarshaller to ProtoStream?

Reply:
I'm not sure what use-case string based is trying to appeal to. If it's simply getting something working as simple as possible, then the `StringMarshaller` is the way to go. Protostream is more flexible, but for text based it has to be JSON and the user has to provide/generate protostream marshallers and a schema, so it's a lot more involved.

Reply:
hey @ryanemerson awesome. this helps clear it up a little for me. I've reworded to include some of this detail. IMO we should include ProtoStream in here too for consistency. also, when you say "Protostream is more flexible" - I take this to mean that you're pretty much restricted to string-based if you use StringMarshaller. that might not make much sense for Hot Rod clients. so if you want text + binary use ProtoStream. but, in this case, text means you must use the application/json mediatype.

does that sound right?

Reply:
> I take this to mean that you're pretty much restricted to string-based if you use StringMarshaller. that might not make much sense for Hot Rod clients.

Exactly.



> so if you want text + binary use ProtoStream. but, in this case, text means you must use the application/json mediatype.

I was actually wrong on this. You can still specify [`text/plain`, `application/json`, 'application/x-java-object', 'application/octet-stream']  media types and transcoding will handle it, as it converts the value to a Java object if possible and then marshalls this using Protostream. XML is not supported though.